util/include/cppmicroservices/util/BundleObjFile.h

```diff
@@ -40,18 +42,39 @@ struct InvalidObjFileException : public std::exception
   std::string m_What;
 };
 
+// Represents the raw bundle resource data.
+// The data buffer is the bits representing a zip file.
+// The data size is the zip file's size in bytes.
+struct RawBundleResources
+{
+  RawBundleResources(std::unique_ptr<void, void(*)(void*)> data, uint64_t dataSize)
```

Comment:
All call sites seem to have a pattern of 
1. allocate a buffer
2. read data into buffer from stream
3. transfer ownership of the buffer to RawBundleResources object

I think you can encapsulate the memory allocation/release inside this class and avoid the repetitive steps at call sites.

example usage ...

```
std::unique_ptr<RawBundleResources> zipData = new RawBundleResources(zipSize);
if(zipData) {
  fs.seekg(fileOffset + section.offset);
  fs.read(zipData.get<char>(), zipSize);
  return zipData;
}
```
instead of 

```
std::unique_ptr<char[]> zipData = std::unique_ptr(new char[zipSize]);
  if(zipData) {
   fs.seekg(fileOffset + section.offset);
   fs.read(zipData.get(), zipSize);
   return std::make_shared<RawBundleResources>( std::move(zipData), zipSize );
 }
```